pkg/operator/controllers/dnsmasq/scripts/99-dnsmasq-restart.gotmpl

```diff
@@ -27,19 +58,37 @@ log_dns_files() {
     log "$(echo -n "/etc/resolv.conf.dnsmasq file metadata: ") $(ls -lZ /etc/resolv.conf.dnsmasq)"
 }
 
-if [[ $interface == eth* && $action == "up" ]] || [[ $interface == eth* && $action == "down" ]] || [[ $interface == enP* && $action == "up" ]] || [[ $interface == enP* && $action == "down" ]]; then
-    log "$action happened on $interface, connection state is now $CONNECTIVITY_STATE"
-    log "Pre dnsmasq restart file information"
-    log_dns_files
-    log "restarting dnsmasq now"
-    if systemctl try-restart dnsmasq --wait; then
-        log "dnsmasq successfully restarted"
-        log "Post dnsmasq restart file information"
-        log_dns_files
-    else
-        log "failed to restart dnsmasq"
+# /etc/resolv.conf.dnsmasq should only be missing under the following conditions
+# - Node is the bootstrapper
+# - dnsmasq hasn't been started yet
+# A node boot will cause eth0 to come up before dnsmasq has been started, causing the lack of /etc/resolv.conf.dnsmasq
+#
+# We restart dnsmasq if /etc/resolv.conf.dnsmasq is missing but dnsmasq is active to restore /etc/resolv.conf.dnsmasq
+#
+# If dnsmasq isn't running, this script won't start dnsmasq. This is due to eth0 coming up/down during the boot process before dnsmasq has been started
+# Preventing an out of order boot sequence or failed boot
+sanity_check() {
+    # state is captured to account for multiple conditions below
+    systemctl is-active dnsmasq --quiet
+    dnsmasq_state=$?
+
+    resolv="/etc/resolv.conf"
+    dnsmasq_resolv="/etc/resolv.conf.dnsmasq"
+    # renvewing the dhcp lease maybe a corrective action to take if /etc/resolv.conf is missing
+    if [[ ! -f $resolv ]]; then
+        abort "$resolv is missing. Sanity check failed."
+    # If this condition is met, dnsmasq will be restarted in main
+    elif [[ ! -f $dnsmasq_resolv ]] && [[ $dnsmasq_state -eq 0 ]]; then
+        log "$dnsmasq_resolv is missing and dnsmasq is active. Something has removed $dnsmasq_resolv. Sanity check failed.
+        Restarting dnsmasq to restore $dnsmasq_resolv"
     fi
-fi
 
-exit 0
+    # dnsmasq won't be running during initial boot, which will trigger this script
+    if [[ ! $dnsmasq_state -eq 0 ]]; then
```

Comment:
Is this something we want to check first in the sanity_check(), rather than last?